### ISSUE TITLE
Error on startup in Godot 3.3 due to music playback

### ISSUE THREAD

**Issue by Calinou** [KOBUGE-Games/jetpaca#28]
**OS:** Fedora 33

After importing the project, running it will crash instantly as soon as the game starts (before any sound starts playing).

The commit of my `3.2` build is https://github.com/godotengine/godot/commit/a1222e435 (so, before the threading modernization).

```
Godot Engine v3.2.4.rc.custom_build.a1222e435 - https://godotengine.org
OpenGL ES 2.0 Renderer: GeForce GTX 1080/PCIe/SSE2
OpenGL ES Batching: ON
 
Level: (1, 1) path: res://stages/world_1/intro.tscn
Level: (1, 2) path: res://stages/world_1/forest_fun.tscn
Level: (1, 3) path: res://stages/world_1/roller_coaster.tscn
Level: (1, 4) path: res://stages/gcon/demo_gcon_004.tscn
Level: (1, 5) path: res://stages/gcon/demo_gcon_006.tscn
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /lib64/libc.so.6(+0x3da60) [0x7f1e91597a60] (??:0)
[2] void memdelete<Thread>(Thread*) (/home/hugo/Documents/Git/godotengine/godot/./core/os/memory.h:117)
[3] _Thread::wait_to_finish() (/home/hugo/Documents/Git/godotengine/godot/core/bind/core_bind.cpp:2837)
[4] MethodBind0R<Variant>::call(Object*, Variant const**, int, Variant::CallError&) (/home/hugo/Documents/Git/godotengine/godot/./core/method_bind.gen.inc:237)
[5] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/home/hugo/Documents/Git/godotengine/godot/core/object.cpp:919)
[6] MessageQueue::_call_function(Object*, StringName const&, Variant const*, int, bool) (/home/hugo/Documents/Git/godotengine/godot/core/message_queue.cpp:250)
[7] MessageQueue::flush() (/home/hugo/Documents/Git/godotengine/godot/core/message_queue.cpp:299)
[8] SceneTree::idle(float) (/home/hugo/Documents/Git/godotengine/godot/scene/main/scene_tree.cpp:523)
[9] Main::iteration() (/home/hugo/Documents/Git/godotengine/godot/main/main.cpp:2108)
[10] OS_X11::run() (/home/hugo/Documents/Git/godotengine/godot/platform/x11/os_x11.cpp:3634)
[11] /home/hugo/Documents/Git/godotengine/godot/bin/godot.x11.tools.64.llvm(main+0x181) [0x2ac4e91] (/home/hugo/Documents/Git/godotengine/godot/platform/x11/godot_x11.cpp:56)
[12] /lib64/libc.so.6(__libc_start_main+0xf2) [0x7f1e915821e2] (??:0)
[13] /home/hugo/Documents/Git/godotengine/godot/bin/godot.x11.tools.64.llvm(_start+0x2e) [0x2ac4c4e] (??:?)
-- END OF BACKTRACE --
[1]    39563 IOT instruction (core dumped)  ~/Documents/Git/godotengine/godot/bin/godot.x11.tools.64.llvm
```

There isn't much more information other than the stack trace – I don't get any errors in the console.

**Edit:** Commenting out the `change_music_to()` call in `shared/music.gd` fixes the game startup:

```gdscript
func _ready():
	pass
	#change_music_to("res://music/bgm.ogg")
```

**Comment by akien-mga** [comment]
That's a duplicate of https://github.com/godotengine/godot/issues/39776, but it makes sense to have it reported here (and we should likely work it around in Jetpaca until that bug gets fixed upstream).

**Comment by akien-mga** [comment]
The crash seems fixed in Godot 3.2.4 RC 3 with the Thread modernization, now we get:
```
ERROR: ~_Thread: Reference to a Thread object was lost while the thread is still running...
   At: core/bind/core_bind.cpp:2847.
```
but no crash.

We should still figure out a proper fix for this, it seems Jetpaca code misuses the Thread API.
